src/main/java/org/broadinstitute/hellbender/utils/read/GATKRead.java

```diff
@@ -730,5 +734,220 @@ default String commonToString() {
             return String.format("%s %s:%d-%d", getName(), getContig(), getStart(), getEnd());
         }
     }
+
+    /**
+     * Returns the offset (0-based index) of the first base in the read that is aligned against the reference.
+     * <p>
+     *     In most cases for mapped reads, this is typically equal to the sum of the size of soft-clipping at the
+     *     beginning of the alignment.
+     * </p>
+     * <p>
+     *     It returns {@code -1} when there is no base against the reference (e.g. the read is unmapped or is all in
+     *     an insertion). Also it will return {@code -1} if it has an empty cigar.
+     * </p>
+     * <p>
+     *     Notice that this index makes reference the base offset as returned by {@link #getBases()}, If you
+     *     are after the first base in the original read considering the strand it maps to you need to use
+     *     {@link #getFirstAlignedReadPosition}
+     * </p>
+     *
+     * @throws IllegalStateException if the read is not marked as unmapped but it does not have a cigar.
+     *
+     * @return a number between {@code -1} and the read length-1.
+     */
+    default int getFirstAlignedBaseOffset() {
```

Comment:
These new methods should probably be in `ReadUtils` rather than in the `GATKRead` interface